### PR TITLE
Generate DockerCluster with kubebuilder

### LINKED ISSUE #137: Add a DockerCluster type
/kind feature

**Describe the solution you'd like**
In line with the AWSCluster object, we will also need a docker specific cluster object. We may not use any fields in it, but it should follow v1alpha1.Machine.ProviderSpec => Infrastructure ProviderMachine

as in, v1alpha1.Cluster.ProviderSpec => InfrastructureProviderCluster

please use Kubebuilder v0.2.0-beta.4 or newer to generate this code

/priority important-soon

### PR DESCRIPTION
**Pull request by ericzhang08**
**What this PR does / why we need it**:
Add a DockerCluster type
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #137 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ var (
 func init() {
 	_ = scheme.AddToScheme(myscheme)
 	_ = v1alpha1.AddToScheme(myscheme)
+	_ = v1alpha1.AddToScheme(myscheme)
```
Comment (chuckha) [inline cmd/manager/main.go]: was this autogenerated? they should already be in the same scheme. Probably a bug in kubebuilder, but you can delete this line

Comment (ericzhang08) [reply cmd/manager/main.go]: OK

File: PROJECT [hunk 2]
```diff
@@ -5,3 +5,6 @@ resources:
 - group: infrastructure
   version: v1alpha1
   kind: DockerMachine
+- group: infrastructure
+  version: v1alpha1
```
Comment (detiber) [inline PROJECT]: Shouldn't this be `v1alpha2`?

@chuckha same for the above DockerMachine.

Comment (chuckha) [reply PROJECT]: oh. Are we matching versions? These didn't exist for v1alpha1, this is the very first version of these types.

**Comment by ericzhang08** [comment]
I have modified  the version of DockerCluster @chuckha @detiber

**Comment by chuckha** [comment]
This looks good to me, I'll assign for final lgtm

/assign @detiber

**Pull request merged**